qip-0016.md

```diff
@@ -0,0 +1,110 @@
+
+Note: When a miner decides to give their coinbase to a smart contract, they are entrusting control of their rewards to that contract. It is important that a miner understands the risk of doing so, because from the perspective of the protocol, the miner technically no longer owns the reward - rather, the coinbase is now a bond and the contract controls the bond on behalf of the miner.
+
+### Smart Contracts, NFTs and more
+
+Ideally, just as a miner can create a bond, so too should a user be able to purchase one from a miner. If user demand for bonds increases, more miners would switch to mining bonds, which would further delay inflation into the future. Conversely, if no users demand bonds, miners can mine bonds and hold them to maturity for an increase in their rewards. 
+
+In order to allow users to purchase bonds, the protocol allows for miners to create tranches, which is effectively a bond position with an interest rate and a maturity date. The miner gives control of this position to a smart contract, and a smart contract mints a tokenized form of this position to the miner, likely constructed as a non-fungible token, or NFT (ERC-1155). When a miner receives this non-fungible token, they can then list it on a decentralized exchange, where users can view various bonds and purchase them in bulk.
+
+A **tranche** is keyed by the following fields: smart contract address, miner address, lockup byte, epoch
+
+A **tranche** has the following information stored in it: balance, unlock height, number of elements, delegate address
+
+To faciliate this EVM smart contract -> tranche database communication there is a precompile smart contract that acts as an API. A smart contract written in Solidity can call this precompile with tightly packed ABI-encoded data, and based on the length of data provided, a certain function is called in the precompile.
+
+| Input Byte Length  | Function               | Expected Input Types                         
```

Comment:
Everything before this point belongs in the "motivation" section. Lots of good explanation of rationale, but the specification section is really just supposed to be the technical details of how to do the lock/unlock, like you start to get at here.